parquet-hadoop/src/main/java/org/apache/parquet/hadoop/CodecFactory.java

```diff
@@ -48,7 +51,7 @@ public class CodecFactory implements CompressionCodecFactory {
   private final Map<CompressionCodecName, BytesCompressor> compressors = new HashMap<>();
   private final Map<CompressionCodecName, BytesDecompressor> decompressors = new HashMap<>();
 
-  protected final Configuration configuration;
```

Comment:
This causes the Iceberg build to break when bumping to 1.14.0-SNAPSHOT:

```
> Task :iceberg-parquet:compileJava FAILED
/Users/fokkodriesprong/Desktop/iceberg/parquet/src/main/java/org/apache/iceberg/parquet/ParquetCodecFactory.java:60: error: cannot find symbol
        codecClass = configuration.getClassLoader().loadClass(codecClassName);
                                  ^
  symbol:   method getClassLoader()
  location: variable configuration of type ParquetConfiguration
/Users/fokkodriesprong/Desktop/iceberg/parquet/src/main/java/org/apache/iceberg/parquet/ParquetCodecFactory.java:62: error: incompatible types: ParquetConfiguration cannot be converted to Configuration
      codec = (CompressionCodec) ReflectionUtils.newInstance(codecClass, configuration);
```

Reply:
Does it make sense to fix this on the iceberg side?

Reply:
Anyway this is a breaking change. We have to fix it before the release.

Reply:
For Iceberg this is not an issue, since we would remove this class with the 1.14 release (that contains https://github.com/apache/parquet-mr/pull/1134): 
https://github.com/apache/iceberg/blob/866021d7d34f274349ce7de1f29d113395e7f28c/parquet/src/main/java/org/apache/iceberg/parquet/ParquetCodecFactory.java#L28-L32

In Iceberg we shade Parquet, so this will not interfere with other libraries that use a different version of Parquet. I'm curious what if we also have this issue on the Spark side (cc @vinooganesh).